drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -519,6 +519,8 @@ static int spi_mcux_dma_rx_load(const struct device *dev, uint8_t *buf,
 	if (buf == NULL) {
 		/* if rx buff is null, then write data to dummy address. */
 		blk_cfg->dest_address = (uint32_t)&dummy_rx_buffer;
+		blk_cfg->source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
```

Comment:
This seems like the correct option regardless of whether the dummy or not is being used

Reply:
I have removed this line as source increment is already 0 based on the transfer type (PERIPHERAL to MEMORY). I had to update the DMA driver as it was looking at `source and destination addr_adj` elements together.

Reply:
makes sense to me